ReactNativeClient/lib/onedrive-api.js

```diff
@@ -73,14 +74,15 @@ class OneDriveApi {
 	}
 
 	async appDirectory() {
-		const r = await this.execJson('GET', '/drive/special/approot');
+		const driveId = this.acctProps_.driveId;
+		const r = await this.execJson('GET', `/me/drives/${driveId}/special/approot`);
 		return `${r.parentReference.path}/${r.name}`;
 	}
 
 	authCodeUrl(redirectUri) {
 		const query = {
 			client_id: this.clientId_,
-			scope: 'files.readwrite offline_access',
+			scope: 'files.readwrite offline_access sites.readwrite.all',
```

Comment:
How will that work with existing apps that have already authorised, but with a different scope?

Reply:
This is no problem because all requests to the OneDrive API also work without the "sites.readwrite.all" permission for OneDrive Personal accounts.  The "sites.readwrite.all" permission is only important for Business accounts. To make sure I also have tested the PR with an OneDrive Personal Account and without the "sites.readwrite.all" permission without problems. This might only be problematic if a user has logged in with an OneDrive Business account in a previous version. In this case the user might think that he only has to upgrade Joplin in order to use his account but this is no true as he also has to log out and log in again before he can use his OneDrive Business account.